iati/home/management/commands/updatestatistics.py

```diff
@@ -0,0 +1,39 @@
+import requests
+import json
+from django.core.management.base import BaseCommand, CommandError
+from home.models import HomePage
+
+
+class Command(BaseCommand):
+    """A command for manage.py that updates the home page statistics.
+    """
+
+    help = 'Update statistics'
+
+    def handle(self, *args, **options):
+        """The default function Django BaseCommand needs to run."""
```

Comment:
I would definitely break this up into a couple further functions...

- lines 18-27 into something like `get_total_num_activities` with docstring to state it queries the Registry, loops over datasets and totals reported number of activities.

- lines 29-36 into something like `get_total_num_publishers` with similar docstring.

Reply:
Ideally we'd unit test these functions, using mocked data from the Registry - we've had some success with the [responses](https://github.com/getsentry/responses) module. There's an [example in action here](https://github.com/IATI/iati.fetch/blob/fetch-dataset/iati/fetch/tests/test_dataset.py#L86-L120) (which tests [this function](
https://github.com/IATI/iati.fetch/blob/fetch-dataset/iati/fetch/__init__.py#L104-L123)).

(This functionality was what the `fetch` part of pyIATI was intended to be about - as started in this code. It never got completed as we focused on other parts of pyIATI, but those links should provide a good basis for testing).

Reply:
Refactoring resolved in ce3c1bc Testing resolved in that and further commits.